app/src/main/java/com/github/jorgecastillo/kotlinandroid/functional/Future.kt

```diff
@@ -1,35 +1,68 @@
 package com.github.jorgecastillo.kotlinandroid.functional
 
+import kategory.*
 import kotlinx.coroutines.experimental.CommonPool
 import kotlinx.coroutines.experimental.Deferred
 import kotlinx.coroutines.experimental.android.UI
 import kotlinx.coroutines.experimental.async
 import kotlinx.coroutines.experimental.launch
 
+typealias FutureKind<A> = HK<Future.F, A>
+
+fun <A> FutureKind<A>.ev(): Future<A> =
+        this as Future<A>
+
 /**
  * Basic future implementation to achieve asynchronicity based on Kotlin coroutines.
  */
-class Future<T> {
+class Future<T> : FutureKind<T> {
```

Comment:
Both AsyncResult. its instances and Future is not something you'd ever have in this project since those are general purpose datatypes. The current Future implementation based on coroutines is in my opinion not efficient and is missing several things you need on a Future for example capturing exceptions. Once IO lands in kategory that would replace `Future` here. Alternatively we can have an impl like we did for bandhook based on kovenant promises wich supports most of the concerns around Futures or Promises.